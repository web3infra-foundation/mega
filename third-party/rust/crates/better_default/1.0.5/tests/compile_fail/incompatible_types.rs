#![allow(dead_code)]

use better_default::Default;

#[derive(Default)]
enum Enum {
    #[default(0: "aaa")]
    Variant(u32, String),

    Variant2,

    Variant3,
}

#[derive(Default)]
enum Enum2 {
    #[default]
    Variant{
        #[default("aaaa")]
        field1: u32,

        field2: String
    },

    Variant2,

    Variant3,
}

#[derive(Default)]
struct Struct {
    #[default("aaaa")]
    field: u32
}

fn main() {}
