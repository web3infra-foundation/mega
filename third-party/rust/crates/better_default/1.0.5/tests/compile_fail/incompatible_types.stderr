error[E0308]: mismatched types
 --> tests/compile_fail/incompatible_types.rs:7:18
  |
5 | #[derive(Default)]
  |          ------- arguments to this enum variant are incorrect
6 | enum Enum {
7 |     #[default(0: "aaa")]
  |                  ^^^^^ expected `u32`, found `&str`
  |
note: tuple variant defined here
 --> tests/compile_fail/incompatible_types.rs:8:5
  |
8 |     Variant(u32, String),
  |     ^^^^^^^

error[E0308]: mismatched types
  --> tests/compile_fail/incompatible_types.rs:19:19
   |
19 |         #[default("aaaa")]
   |                   ^^^^^^ expected `u32`, found `&str`

error[E0308]: mismatched types
  --> tests/compile_fail/incompatible_types.rs:32:15
   |
32 |     #[default("aaaa")]
   |               ^^^^^^ expected `u32`, found `&str`
